marshal/claim.go

```diff
@@ -501,9 +501,13 @@ func (c *claim) healthCheck() bool {
 		return true
 	}
 
-	// If the consumer is moving faster than the partition, consider it healthy. This case
-	// is true in the standard catching up from behind case.
-	if partitionVelocity <= consumerVelocity {
+	// At this point we know the consumer is NOT PRESENTLY caught up or predicted to catch
+	// up in the next two heartbeats.
+
+	// If the consumer is moving as fast or faster than the partition and the consumer is
+	// at least moving, consider it healthy. This case is true in the standard catching
+	// up from behind case.
+	if partitionVelocity < consumerVelocity || (partitionVelocity <= 0 && consumerVelocity > 0) {
```

Comment:
It should never be less than zero in any situation I can think of. And, uh... yeah, tests still pass if I remove the first half of the statement. I'm worried that means the test isn't catching something.

Updated, there's a new commit that adds a test that shows why we need both sections of the if statement: i.e., the "consumer is catching up but is presently far behind" state.